pywbem/_logging.py

```diff
@@ -0,0 +1,402 @@
+            # All members of this tuple are just viewing except detail
+            # that is used by individual loggers
+            cls.loggers[logger_name] = (log_detail_level, log_level,
+                                        log_dest, log_filename)
+
+    @classmethod
+    def get_logger(cls, logger_name):
+        """
+            Get the name of a logger that has been defined.
+        """
+        if logger_name in cls.loggers:
+            return cls.loggers[logger_name][0]
+        return None
+
+    @classmethod
+    def get_logger_info(cls, logger_name):
+        """Get information about a logger by name"""
+        if logger_name in cls.loggers:
+            t = cls.loggers[logger_name]
+            return 'comp=%s dest=%s detail=%s log_level=%s ' \
+                   'file=%s' % (logger_name, t[3], t[0], t[2], t[3])
+        return('%s not defined.' % logger_name)
+
+    @classmethod
+    def _parse_log_specs(cls, log_spec_str):
+        """
+        Parse a complete cmd line log specification that is in the
+        format defined in PywbemLoggers.create_logger
+
+          Parameters:
+            log_spec_str (:term:`string`) containing the log spec string
+
+          Return:
+            Dictionary containing the parsed information where each entry is
+            log_comp (detail_level, log_level,  dest)
+
+          Exception:
+            ValueError if parse fails
+        """
+        log_specs_dict = dict()
+        log_specs = log_spec_str.split(',')
+        for log_spec in log_specs:
+            try:
+                log_component, log_data = log_spec.split("=", 1)
+            except ValueError:
+                log_component = log_spec
+                log_data = ''
+
+            log_values = log_data.split(":")
+
+            if not log_component:
+                raise ValueError("Log component name required in %s" % log_spec)
+
+            # cvt empty strings to None
+            log_values = [None if x == '' else x for x in log_values]
+            # expand to full size if not all values supplied
+            while len(log_values) < 3:
+                log_values.append(None)
+
+            if len(log_values) > 3:
+                raise ValueError("Invalid log detail. %s too many "
+                                 "components."
+                                 % log_spec)
+            log_specs_dict[log_component] = tuple(log_values)
+        return log_specs_dict
+
+
+def get_logger(name):
+    """
+    Return a :class:`~py:logging.Logger` object with the specified name.
+
+    A :class:`~py:logging.NullHandler` handler is added to the logger if it
+    does not have any handlers yet. This prevents the propagation of log
+    requests up the Python logger hierarchy, and therefore causes this package
+    to be silent by default.
+    """
+    logger = logging.getLogger(name)
+    if not logger.handlers:
+        logger.addHandler(logging.NullHandler())
+    return logger
```

Comment:
If the recommended approach is to use `PywbemLoggers.create_logger()` to create a logger, why do we have the `get_logger()` function in addition?